### PR TITLE
Feat: add Gradient Illumination augmentations (gaussian)

### PR DESCRIPTION
**Pull request by vgilabert94**
#### Changes
Add Random illumination augmentation. 
- [x] Gaussian
- [ ] Radial 
- [ ] Linear
- Etc...

Gaussian illumination examples:
image

Please review gaussian illumination (strcuture, names, parameters, generator, docs, tests), when it is correct, I will add more transformations along these guidelines.

related to https://github.com/kornia/kornia/issues/2721
related to https://github.com/kornia/kornia/issues/941

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/augmentation/_2d/intensity/gaussian_illumination.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        # Validation and initialization of amount parameter.
+        if isinstance(gain, (tuple, float)):
+            if isinstance(gain, float):
+                gain = (gain, gain)
+            elif len(gain) == 1:
+                gain = (gain[0], gain[0])
+            elif len(gain) > 2 or len(gain) <= 0:
+                raise ValueError(
+                    "The length of amount must be greater than 0 \
```
Comment (johnnv1) [inline kornia/augmentation/_2d/intensity/gaussian_illumination.py:77]: it can be a float too

File: kornia/augmentation/random_generator/_2d/gaussian_illumination.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+import torch
+
+from kornia.augmentation.random_generator.base import RandomGeneratorBase, UniformDistribution
+from kornia.augmentation.utils import _adapted_rsampling, _common_param_check, _range_bound
+from kornia.core import Tensor
+from kornia.utils import _extract_device_dtype
+
+
+def gaussian(window_size, sigma, center=None):
```
Comment (edgarriba) [inline kornia/augmentation/random_generator/_2d/gaussian_illumination.py:11]: there's an existing function https://github.com/kornia/kornia/blob/main/kornia/filters/kernels.py#L62

File: kornia/augmentation/random_generator/_2d/gaussian_illumination.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            device=_device, dtype=_dtype
+        )
+        sign = torch.where(
+            _adapted_rsampling((batch_size,), self.sign_sampler, same_on_batch) >= 0.0,
+            torch.tensor(1),
+            torch.tensor(-1),
+        ).to(device=_device, dtype=_dtype)
+
+        gradient = torch.zeros(batch_shape)
+        for batch in range(batch_size):
```
Comment (edgarriba) [inline kornia/augmentation/random_generator/_2d/gaussian_illumination.py:110]: why it cannot be done in batch ?

File: kornia/augmentation/random_generator/_2d/gaussian_illumination.py (hunk 2 above)
Comment (vgilabert94) [reply kornia/augmentation/random_generator/_2d/gaussian_illumination.py:11]: Yes, I copied it from there but adding a parameter of center.
If you think it's ok to add center we can merge this two functions if it does not change normal behaviour.

File: kornia/augmentation/random_generator/_2d/gaussian_illumination.py (hunk 3 above)
Comment (vgilabert94) [reply kornia/augmentation/random_generator/_2d/gaussian_illumination.py:110]: I'm not sure, but in order to do it in batch I have to be able to generate the gaussians in batch and I didn't know how to do it. If you can help me, we will integrate it.

File: kornia/augmentation/random_generator/_2d/gaussian_illumination.py (hunk 2 above)
Comment (johnnv1) [reply kornia/augmentation/random_generator/_2d/gaussian_illumination.py:11]: +1 to add support to center on the gaussian kernel function. If it keeps back-compatibility, and add this feature, I do not see why not 😄

File: kornia/filters/kernels.py [hunk 4]
```diff
@@ -80,7 +87,16 @@ def gaussian(
     KORNIA_CHECK_SHAPE(sigma, ["B", "1"])
     batch_size = sigma.shape[0]
 
-    x = (torch.arange(window_size, device=sigma.device, dtype=sigma.dtype) - window_size // 2).expand(batch_size, -1)
+    if center is None:
```
Comment (johnnv1) [inline kornia/filters/kernels.py:90]: can you add a specific case for it under kernels tests?

File: kornia/filters/kernels.py [hunk 5]
```diff
@@ -60,13 +60,20 @@ def normalize_kernel2d(input: Tensor) -> Tensor:
 
 
 def gaussian(
-    window_size: int, sigma: Tensor | float, *, device: Optional[Device] = None, dtype: Optional[Dtype] = None
+    window_size: int,
+    sigma: Tensor | float,
+    *,
+    center: Tensor | float = None,
```
Comment (edgarriba) [inline kornia/filters/kernels.py:66]: To be more accurated with terminology, could we rename this as `mean`?

File: kornia/filters/kernels.py (hunk 4 above)
Comment (edgarriba) [inline kornia/filters/kernels.py:90]: `center = center or float(window_size // 2)a
For one line expression

**Review by edgarriba — APPROVED** [review]

File: kornia/filters/kernels.py (hunk 4 above)
Comment (vgilabert94) [reply kornia/filters/kernels.py:90]: There are no test for gaussian function, i assume that they are inside get_gaussian_kernel1d. 
I will have to add "mean" argument to get_gaussian_kernel1d as well, won't I? and what about get_gaussian_kernel2d and get_gaussian_kernel2d?

Comment (johnnv1) [reply kornia/filters/kernels.py:90]: ouch, we are missing some tests. I would say to just add a simple test for `gaussian`... I'm not sure if "mean" is relevant for the internal kernel generators

**Review by johnnv1 — APPROVED** [review]

File: tests/augmentation/test_augmentation.py [hunk 6]
```diff
@@ -3291,6 +3292,83 @@ def test_same_on_batch(self, device, dtype):
         self.assert_close(output_tensor[0], output_tensor[1])
 
 
+class TestRandomGaussianIllumination(BaseTester):
+    @pytest.mark.parametrize(
+        "expected",
+        [
+            torch.tensor(
```
Comment (johnnv1) [inline tests/augmentation/test_augmentation.py:3299]: put it directly in the function, instead of `parametrize`

File: tests/filters/test_gaussian.py [hunk 7]
```diff
@@ -14,6 +15,35 @@
...
+            11,
+            11.0,
+            3.0,
+            torch.tensor([[0.0926, 0.0946, 0.0957, 0.0961, 0.0957, 0.0946, 0.0926, 0.0900, 0.0867, 0.0828, 0.0785]]),
+        ),
+    ],
+)
+def test_gaussian(window_size, sigma, mean, expected):
+    result = gaussian(window_size, sigma, mean=mean)
+    assert_close(result.sum(), expected.sum())
```
[the remark this answers is not in the document]
Comment (johnnv1) [reply tests/filters/test_gaussian.py:44]: To avoid creating tensors on tests suite build (or discovery?), and also be able to test with device and dtype...

why are you asserting the sums? I think it may fail because of the expected precision, if this is the case we can use `atol`/`rtol` to change the assert_close tolerance

File: tests/augmentation/test_augmentation.py [hunk 8]
```diff
@@ -3291,6 +3292,83 @@ def test_same_on_batch(self, device, dtype):
...
+            RandomGaussianIllumination(sigma=(0.01, 2))
+
+        with pytest.raises(Exception, match="center of gaussian value must be between 0 and 1."):
+            RandomGaussianIllumination(center=(0.01, 2))
+
+        with pytest.raises(Exception, match="sign of gaussian value must be between -1 and 1."):
+            RandomGaussianIllumination(sign=(0.01, 2))
+
+    @pytest.mark.parametrize("batch_shape", [1, 3, 3, 5])
+    @pytest.mark.parametrize("channel_shape", [1, 1, 2, 3])
```
Comment (johnnv1) [inline tests/augmentation/test_augmentation.py:3357]:
```suggestion
    @pytest.mark.parametrize("batch_shape", [1, 3, 5])
    @pytest.mark.parametrize("channel_shape", [1, 2, 3])
```
using separated `parametrize` will create a product between the cases, instead of being aligned/sequential... if you want to control the value pairs, you will need something like

```python
    @pytest.mark.parametrize("channel_shape,batch_shape", [(1, 1), (3, 2), (5, 3)])
```

File: kornia/augmentation/_2d/intensity/gaussian_illumination.py [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+        self,
+        input: Tensor,
+        params: Dict[str, Tensor],
+        flags: Dict[str, Any],
+        transform: Optional[Tensor] = None,
+    ) -> Tensor:
+        r"""Apply random gaussian gradient illumination to the input image."""
+        KORNIA_CHECK(len(input.shape) in (3, 4), "Wrong input dimension.")
+        if len(input.shape) == 3:
+            input = input[None, :, :, :]
```
Comment (johnnv1) [inline kornia/augmentation/_2d/intensity/gaussian_illumination.py:154]: it's automatically handled by the base API, so you don't need to check or try to handle the input shape

```suggestion
```

**Review by johnnv1 — APPROVED** [review]

**Comment by johnnv1** [comment]
The failing test is unrelated to this patch. Just an issue with an asset on the tutorial dataset. Thank you for this patch 😄
